Graphics_Engine/Modify/ApplyGradientOptions.cs

```diff
@@ -38,12 +38,14 @@ public static partial class Modify
         /****           Public Methods                  ****/
         /***************************************************/
 
+        [PreviousVersion("5.1", "BH.Engine.Graphics.Modify.ApplyGradientOptions(BH.oM.Graphics.Colours.GradientOptions, System.Collections.Generic.IEnumerable<System.Double>, System.String)")]
         [Description("Sets up the properties of a GradientOptions object for usage.")]
         [Input("gradientOptions", "GradientOptions object to modify.")]
         [Input("allValues", "The values to set gradient auto range from. Optional if range is already set.")]
+        [Input("gradientBoundsWarning", "If true, a warning will be raised if the bounds have been manually set and any of the provided values are outside of the bounds domain.")]
         [Input("defaultGradient", "Sets which gradient to use as default if no gradient is already set. Defaults to BlueToRed.")]
         [Output("gradientOptions", "A GradientOptions object which is ready for usage.")]
-        public static GradientOptions ApplyGradientOptions(this GradientOptions gradientOptions, IEnumerable<double> allValues = null, string defaultGradient = "BlueToRed")
+        public static GradientOptions ApplyGradientOptions(this GradientOptions gradientOptions, IEnumerable<double> allValues = null, bool gradientBoundsWarning = true, string defaultGradient = "BlueToRed")
```

Comment:
I think that the option `gradientBoundsWarning` could be better placed as the last input parameter, being only of contextual utility? At least that is what I do when I have similar cases

Reply:
I actually put it last when first adding the parameter, but then moved it in front of the default gradient. My reason for putting it there was that I thought it was more likely that you would want to change whether to display the warning or not than it was likely that you would want to change the default gradient, hence leaving the default gradient last.

Can move it to the last position if you disagree with this.

Reply:
After our conversation, my understanding is that the proper solution would be:
- `defaultGradient` should be a default value of `GradientOptions` object
- the method should accept a null `GradientOptions` object or one with a null `DefaultGradient` property, and instantiate it if it is (therefore getting the default value populated)
- The first point is difficult to achieve due to the current mechanism of where the default values are pulled from (datasets)

I am not sure I summarised this correctly 😄 
Anyway, because this is not important, nothing needs to be done here in this PR.
I leave it to you if we want to raise an issue with the points above to be solved in the future or not.

Reply:
Will leave it as it is for now!